test/integration/api/order.spec.js

```diff
@@ -656,6 +656,246 @@ test('updates an order move', async (t) => {
   t.is(orderMove.metadata.test, true)
 })
 
+test('simulates a payment process', async (t) => {
+  const authorizationHeaders = await getAccessTokenHeaders({
+    t,
+    permissions: [
+      'asset:list:all',
+      'asset:create:all',
+      'transaction:transition:all',
+      'order:read:all',
+      'order:create:all',
+      'orderMove:create:all',
+    ]
+  })
+
+  const sellingAssetTypeId = 'typ_MGsfQps1I3a1gJYz2I3a'
+  const ownerId = 'usr_em9SToe1nI01iG4yRnHz'
+
+  // /////////////// //
+  // ASSETS CREATION //
+  // /////////////// //
+
+  const { body: asset1 } = await request(t.context.serverUrl)
+    .post('/assets')
+    .send({
+      name: 'Chevrolet',
+      ownerId,
+      assetTypeId: sellingAssetTypeId,
+      quantity: 5,
+      currency: 'USD',
+      price: 50000,
+    })
+    .set(authorizationHeaders)
+    .expect(200)
+
+  const { body: asset2 } = await request(t.context.serverUrl)
+    .post('/assets')
+    .send({
+      name: 'Toyota',
+      ownerId,
+      assetTypeId: sellingAssetTypeId,
+      quantity: 10,
+      currency: 'USD',
+      price: 40500,
+    })
+    .set(authorizationHeaders)
+    .expect(200)
+
+  // /////////////////// //
+  // USER AUTHENTICATION //
+  // /////////////////// //
+
+  const { body: loginObject } = await request(t.context.serverUrl)
+    .post('/auth/login')
+    .set({
+      'x-platform-id': t.context.platformId,
+      'x-stelace-env': t.context.env
+    })
+    .send({
+      username: 'user',
+      password: 'user'
+    })
+    .expect(200)
+
+  const userHeaders = {
+    'x-platform-id': t.context.platformId,
+    'x-stelace-env': t.context.env,
+    authorization: `${loginObject.tokenType} ${loginObject.accessToken}`
+  }
+
+  let { body: user } = await request(t.context.serverUrl)
+    .get(`/users/${loginObject.userId}`)
+    .set(userHeaders)
+    .expect(200)
+
+  // //////// //
+  // SCENARIO //
+  // //////// //
+
+  // update cart without creating any transaction
+  user = await updateCart(user, { assetId: asset1.id, quantity: 4 })
+  t.is(getCart(user).length, 1)
+
+  user = await updateCart(user, { assetId: asset2.id, quantity: 2 })
+  t.is(getCart(user).length, 2)
+
+  user = await updateCart(user, { assetId: asset1.id, quantity: 0 })
+  t.is(getCart(user).length, 1)
+
+  user = await updateCart(user, { assetId: asset1.id, quantity: 5 })
+  t.is(getCart(user).length, 2)
+
+  // get information before payment
+  const previewedTransactions = await getPreviewedTransactions(user)
+  previewedTransactions.forEach(preview => {
+    t.true(_.isString(preview.assetId))
+    t.true(_.isNumber(preview.quantity))
+    t.true(_.isNumber(preview.unitPrice))
+    t.true(_.isNumber(preview.value))
+  })
```

Comment:
~Is there some specific reason to test transaction preview objects this here in `order.spec.js`?~

As mentioned in another comment, I suppose it’s because you test the whole shopping cart process, not just the Order API despite the location of this test.